jwst/assign_wcs/miri.py

```diff
@@ -74,53 +74,128 @@ def imaging_distortion(input_model, reference_files):
+
+    # Setup the frames.
     detector = cf.Frame2D(name='detector', axes_order=(0, 1), unit=(u.pix, u.pix))
+
     focal_spatial = cf.Frame2D(name='focal', axes_order=(0, 1), unit=(u.arcmin, u.arcmin))
-    #sky = cf.CelestialFrame(reference_frame=coord.ICRS())
     spec = cf.SpectralFrame(name='wavelength', axes_order=(2,), unit=(u.micron,), axes_names=('lambda',))
     focal = cf.CompositeFrame([focal_spatial, spec])
 
-    ref = fits.open(reference_files['specwcs'])
-    ldata = ref[1].data
-    if input_model.meta.exposure.type.lower() == 'mir_lrs-fixedslit':
-        zero_point = ref[1].header['imx'], ref[1].header['imy']
-    elif input_model.meta.exposure.type.lower() == 'mir_lrs-slitless':
-        #zero_point = ref[1].header['imysltl'], ref[1].header['imxsltl']
-        #zero point in reference file is wrong
-        # This should eb moved eventually to the reference file.
-        zero_point = [35, 442]#[35, 763] # account for subarray
-    lrsdata = np.array([l for l in ldata])
+    # Determine the distortion model.
+    distortion = imaging_distortion(input_model, reference_files)
+
+    # Load and process the reference data.
+    with fits.open(reference_files['specwcs']) as ref:
+        lrsdata = np.array([l for l in ref[1].data])
+
+        # Get the zero point from the reference data.
+        # The zero_point is X, Y  (which should be COLUMN, ROW)
+        if input_model.meta.exposure.type.lower() == 'mir_lrs-fixedslit':
+            zero_point = ref[1].header['imx'], ref[1].header['imy']
+        elif input_model.meta.exposure.type.lower() == 'mir_lrs-slitless':
+            zero_point = ref[1].header['imxsltl'], ref[1].header['imysltl']
+
+    # Create the domain
     x0 = lrsdata[:, 3]
-    x1 = lrsdata[:, 5]
     y0 = lrsdata[:, 4]
+    x1 = lrsdata[:, 5]
+
     domain = [{'lower': x0.min() + zero_point[0], 'upper': x1.max() + zero_point[0]},
               {'lower': (y0.min() + zero_point[1]), 'upper': (y0.max() + zero_point[1])}
               ]
-    log.info("Setting domain to {0}".format(domain))
+
+    # Find the ROW of the zero point which should be the [1] of zero_point
+    row_zero_point = zero_point[1]
+
+    # Compute the v2v3 to sky.
+    tel2sky = pointing.v23tosky(input_model)
+
+    # Compute the V2/V3 for each pixel in this row
+    # x.shape will be something like (1, 388)
+    y, x = np.mgrid[row_zero_point:row_zero_point+1, 0:input_model.data.shape[1]]
+
+    radec = distortion | tel2sky
+    radec = np.array(radec(x, y))[:, 0, :]
+
+    ra_full = np.matlib.repmat(radec[0], domain[1]['upper']+1-domain[1]['lower'], 1)
+    dec_full = np.matlib.repmat(radec[1], domain[1]['upper']+1-domain[1]['lower'], 1)
+
+    ra_t2d = models.Tabular2D(lookup_table=ra_full)
+    dec_t2d = models.Tabular2D(lookup_table=dec_full)
+
+    # Create the model transforms.
     lrs_wav_model = jwmodels.LRSWavelength(lrsdata, zero_point)
-    ref.close()
+
+    # Incorporate the small rotation
     angle = np.arctan(0.00421924)
```

Comment:
I agree the angle should be in the reference file. Right now it comes from the report which accompanies the old reference file. I have not looked at the new report. 
I am not sure this rotation should be applied though. It was my interpretation of the report at the time that it should be applied but this was never confirmed/denied by the team.
We should look at the report that came with the latest reference file and at a recent response from the IDT (which I;ll forward separately).